### PR TITLE
`azurerm_subnet` - the private_endpoint_network_policies_enabled property has been deprecated in favour of the `private_endpoint_network_policies` property

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description
Currently, azurerm_subnet has supported the ["privateEndpointNetworkPolicies"](https://github.com/Azure/azure-rest-api-specs/blob/99686ad45c380cf8fbcf7811db9c984a0e9d6639/specification/network/resource-manager/Microsoft.Network/stable/2023-09-01/virtualNetwork.json#L1602) property and it is marked as bool type in TF. But it's not expected since this property is expanded with new values and now it supports these values "Enabled"/"Disabled"/"NetworkSecurityGroupEnabled"/"RouteTableEnabled" in Swagger. So we have to convert it from bool to string to support more values in TF. This feature request is from Service team.

image

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_subnet` - convert the `privateEndpointNetworkPolicies` property from bool to string to support more values

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/network/subnet_data_source.go [hunk 1]
```diff
@@ -83,6 +84,11 @@ func dataSourceSubnet() *pluginsdk.Resource {
 	}
 
 	if !features.FourPointOhBeta() {
+		resource.Schema["private_endpoint_network_policies_enabled"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		}
```
Comment (katbyte) [inline internal/services/network/subnet_data_source.go:90]: we should add a deprecation message here?

**Review by katbyte — CHANGES_REQUESTED** [review]
Aside from one commment i think this looks good

File: internal/services/network/subnet_data_source.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/network/subnet_data_source.go:91]: Added

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for the comment. I updated PR. Please take another look.

File: internal/services/network/subnet_data_source.go [hunk 2]
```diff
@@ -83,6 +84,12 @@ func dataSourceSubnet() *pluginsdk.Resource {
 	}
 
 	if !features.FourPointOhBeta() {
+		resource.Schema["private_endpoint_network_policies_enabled"] = &pluginsdk.Schema{
+			Type:       pluginsdk.TypeBool,
+			Computed:   true,
+			Deprecated: "This is deprecated in favor of `private_endpoint_network_policies`",
```
Comment (stephybun) [inline internal/services/network/subnet_data_source.go:90]: Can you please specify that the property will be removed in v4.0 of the provider 
```suggestion
			Deprecated: "This property has been superseded by `private_endpoint_network_policies` and will be removed in v4.0 of the AzureRM Provider.",
```

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/network/subnet_data_source.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/network/subnet_data_source.go]: Updated

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comment. I updated PR. Please take another look.

**Review by katbyte — APPROVED** [review]
LGTM 🦕

**Pull request merged**